app/models/host_initiator.rb

```diff
@@ -0,0 +1,18 @@
+class HostInitiator < ApplicationRecord
+  include ProviderObjectMixin
+  include SupportsFeatureMixin
+  include CustomActionsMixin
+
+  belongs_to :ext_management_system, :foreign_key => :ems_id
+  belongs_to :physical_storage, :inverse_of => :host_initiators
+
+  has_many :san_addresses, :as => :owner, :dependent => :destroy
+
+  virtual_total :v_total_addresses, :san_addresses
+
+  def self.class_by_ems(ext_management_system)
+    # TODO(lsmola) taken from Orchestration stacks, correct approach should be to have a factory on ExtManagementSystem
+    # side, that would return correct class for each provider
+    ext_management_system && ext_management_system.class::HostInitiator
+  end
```

Comment:
I don't think we need class_by_ems, since we don't expect people to be sublcalssing this in their providers.

Reply:
removed